cmake/detect-malloc.cmake

```diff
@@ -0,0 +1,32 @@
+# Neither Windows nor macOS support ISO C11 `aligned_alloc`.
```

Comment:
This is almost the same thing as before but in comment form. Where when a new platform/sdk comes along are there going to be new commits for everybody wanting to add in their own URL for the docs.

Reply:
If these new "platform/sdk" support portable functions (e.g. Fuchsia), then no action is required in the build system.